src/yunohost/tools.py

```diff
@@ -534,9 +534,21 @@ def _list_upgradable_apps():
         app_dict = app_info(app_id, raw=True)
 
         if app_dict["upgradable"] == "yes":
+
+            current_version = app_dict.get("version", "?")
+            current_commit = app_dict.get("status", {}).get("remote", {}).get("revision", "?")[:7]
```

Comment:
If your `{}` defaults are returned from `.get` then the `[:7]` will throw an `IndexError`?

We need a `try/except` block here or?

Reply:
Uh nope ? The last default return is `"?"` so it will simply return `"?"`

Try this in an interactive console : 

```python
app_dict = {}
app_dict.get("status", {}).get("remote", {}).get("revision", "?")[:7]
# Returns '?'
```

Reply:
Yes, slice never raise an IndexError in python :)

Reply:
Ah, a slice, not an index! Well, there is a lot happening on that line :) OK, cool!